hedera-mirror-rest/service/tokenService.js

```diff
@@ -18,20 +18,69 @@
  * ‍
  */
 
-import _ from 'lodash';
-import {Token} from '../model';
+import {TokenAccount, TokenBalance, AccountBalanceFile} from '../model';
+import BaseService from './baseService';
+import {OrderSpec} from '../sql';
 
 /**
  * Token retrieval business logic
  */
-class TokenService {
-  static tokenByIdQuery = `select *
-                           from ${Token.tableName}
-                           where ${Token.TOKEN_ID} = $1`;
+class TokenService extends BaseService {
+  static tokenRelationshipsQuery = `
+        select ${TokenAccount.getFullName(TokenAccount.AUTOMATIC_ASSOCIATION)},
+               ${TokenAccount.getFullName(TokenAccount.CREATED_TIMESTAMP)},
+               ${TokenAccount.getFullName(TokenAccount.FREEZE_STATUS)},
+               ${TokenAccount.getFullName(TokenAccount.KYC_STATUS)},
+               ${TokenAccount.getFullName(TokenAccount.TOKEN_ID)},
+               ${TokenBalance.getFullName(TokenBalance.BALANCE)}
```

Comment:
an account may associate itself with a token after the latest balance snapshot at time `T1`, then the `balance` f rom `token_balance` table can be `null`, since later we will track live token balance and no longer need the left join with `token_balance` table, we should set it to 0

Also need a test case

```suggestion
               coalesce(${TokenBalance.getFullName(TokenBalance.BALANCE)}, 0) balance
```

Reply:
This is a test for that scenario. token-created-after-last-balance.json